src/ILCompiler.Compiler/src/Compiler/PrecomputedMetadataManager.cs

```diff
@@ -0,0 +1,365 @@
+        {
+            return _loadedMetadata.Value.LocalMetadataModules;
+        }
+
+        public override bool IsReflectionBlocked(MetadataType type)
+        {
+            return type.HasCustomAttribute("System.Runtime.CompilerServices", "ReflectionBlockedAttribute");
+        }
+
+        protected override void ComputeMetadata(out byte[] metadataBlob, out List<MetadataMapping<MetadataType>> typeMappings, out List<MetadataMapping<MethodDesc>> methodMappings, out List<MetadataMapping<FieldDesc>> fieldMappings)
+        {
+            MetadataLoadedInfo loadedMetadata = _loadedMetadata.Value;
+            metadataBlob = _metadataBlob;
+            typeMappings = loadedMetadata.AllTypeMappings;
+            methodMappings = loadedMetadata.MethodMappings;
+            fieldMappings = loadedMetadata.FieldMappings;
+        }
+
+        private Dictionary<MethodDesc, MethodDesc> LoadDynamicInvokeStubs()
+        {
+            MetadataType typeWithMetadataMappings = (MetadataType)_metadataDescribingModule.GetTypeByCustomAttributeTypeName(MetadataMappingTypeName);
+            Dictionary<MethodDesc, MethodDesc> dynamicInvokeMapTable = new Dictionary<MethodDesc, MethodDesc>();
+            MethodDesc dynamicInvokeStubDescriptorMethod = typeWithMetadataMappings.GetMethod("DynamicInvokeStubs", null);
+
+            if (dynamicInvokeStubDescriptorMethod == null)
+                return dynamicInvokeMapTable;
+
+            ILProvider ilProvider = new ILProvider(null);
+            ILStreamReader il = new ILStreamReader(ilProvider.GetMethodIL(dynamicInvokeStubDescriptorMethod));
+            // structure is 
+            // REPEAT N TIMES    
+            //ldtoken method
+            //ldtoken dynamicInvokeStubMethod
+            //pop
+            //pop
+            while (true)
+            {
+                if (il.TryReadRet()) // ret
+                    break;
+
+                MethodDesc method = il.ReadLdtokenAsTypeSystemEntity() as MethodDesc;
+                MethodDesc dynamicMethodInvokeStub = il.ReadLdtokenAsTypeSystemEntity() as MethodDesc;
+                il.ReadPop();
+                il.ReadPop();
+
+                if ((method != null) && (dynamicMethodInvokeStub != null))
+                {
+                    dynamicInvokeMapTable[method] = dynamicMethodInvokeStub;
+                }
+                else
+                {
+                    throw new BadImageFormatException();
+                }
+            }
+
+            return dynamicInvokeMapTable;
+        }
+
+        /// <summary>
+        /// Is there a reflection invoke stub for a method that is invokable?
+        /// </summary>
+        public override bool HasReflectionInvokeStub(MethodDesc method)
+        {
+            return GetReflectionInvokeStub(method) != null;
+        }
+
+        private MethodDesc InstantiateDynamicInvokeMethodForMethod(MethodDesc dynamicInvokeMethod, MethodDesc methodToInvoke)
```

Comment:
Is there a reason why we don't want to share this logic with what's in `GetReflectionInvokeStub`?